src/trackedit/internal/trackeditactionscontroller.cpp

```diff
@@ -226,61 +257,98 @@ void TrackeditActionsController::doGlobalCopy()
 }
 
 void TrackeditActionsController::doGlobalCut()
+{
+    dispatcher()->dispatch(CUT_LEAVE_GAP_CODE);
+}
+
+void TrackeditActionsController::doGlobalCutLeaveGap()
 {
     if (selectionController()->timeSelectionIsNotEmpty()) {
-        dispatcher()->dispatch(RANGE_SELECTION_CUT_CODE);
+        auto selectedTracks = selectionController()->selectedTracks();
+        secs_t selectedStartTime = selectionController()->dataSelectedStartTime();
+        secs_t selectedEndTime = selectionController()->dataSelectedEndTime();
+
+        dispatcher()->dispatch(SPLIT_CUT_SELECTED,
+                               ActionData::make_arg3<TrackIdList, secs_t, secs_t>(selectedTracks, selectedStartTime, selectedEndTime));
         return;
     }
 
     if (selectionController()->selectedClips().empty()) {
+        interactive()->error(muse::trc("trackedit", "No audio selected"),
```

Comment:
if we take into account that the doGlobal method determine the context, then I think the error should be at the end of the method

Reply:
@Eism I don't really understand your point here. Could you explain it a bit?